app/controllers/CannotClaimExpenseController.scala

```diff
@@ -18,24 +18,44 @@ package controllers
 
 import config.FrontendAppConfig
 import controllers.actions._
+import controllers.mergedJourney.routes.MergedJourneyController
+import models.mergedJourney.ClaimUnsuccessful
+
 import javax.inject.Inject
 import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 import views.html.CannotClaimExpenseView
 
+import scala.concurrent.{ExecutionContext, Future}
+
 class CannotClaimExpenseController @Inject()(
                                               override val messagesApi: MessagesApi,
+                                              sessionRepository: SessionRepository,
                                               identify: UnauthenticatedIdentifierAction,
                                               getData: DataRetrievalAction,
                                               requireData: DataRequiredAction,
                                               val controllerComponents: MessagesControllerComponents,
                                               view: CannotClaimExpenseView,
-                                              frontendAppConfig: FrontendAppConfig
-                                     ) extends FrontendBaseController with I18nSupport {
+                                     )(implicit val ec: ExecutionContext,
+                                        appConfig: FrontendAppConfig) extends FrontendBaseController with I18nSupport {
 
-  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData) {
-    implicit request =>
-      Ok(view(frontendAppConfig.govUkUrl))
+  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    if (appConfig.mergedJourneyEnabled) {
+      request.identifier match {
+        case id: Authed =>
+          sessionRepository.getMergedJourney(id.internalId).map {
+            case Some(journeyConfig) =>
+              Ok(view(Some(MergedJourneyController.mergedJourneyContinue(journey="fre", status=ClaimUnsuccessful).url)))
```

Comment:
Instead of doing this you can just check if the userAsnwers.isMergedJourney == true
That flag is already blocked by the feature switch.